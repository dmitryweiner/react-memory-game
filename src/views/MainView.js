import React, { Component } from "react";
import Button from "@material-ui/core/Button";
import Statistics from "../components/Statistics";

export default class MainView extends Component {
  render() {
    return (
      <>
        <h4>Игра на запоминание чисел</h4>
        <p>
          Вам будет предложено запомнить некоторые числа. После этого нужно будет их воспроизвести,
          не обязательно в том же порядке, за фиксированное время.
        </p>
        {this.props.stats.games > 0 && <Statistics {...this.props.stats} />}
        <Button variant="contained" color="primary" onClick={() => this.props.handleTabs(1)}>
          Играть!
        </Button>
      </>
    );
  }
}
