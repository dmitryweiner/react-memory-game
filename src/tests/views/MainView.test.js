import React from "react";
import renderer from "react-test-renderer";
import MainView from "../../views/MainView";

test("MainView", () => {
  const component = renderer.create(<MainView />).toJSON();
  expect(component).toMatchSnapshot();
});
