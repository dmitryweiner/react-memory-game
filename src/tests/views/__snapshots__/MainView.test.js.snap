// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainView 1`] = `
Array [
  <h4>
    Игра на запоминание чисел
  </h4>,
  <p>
    Вам будет предложено запомнить некоторые числа. После этого нужно будет их воспроизвести, не обязательно в том же порядке, за фиксированное время.
  </p>,
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Играть!
    </span>
  </button>,
]
`;
